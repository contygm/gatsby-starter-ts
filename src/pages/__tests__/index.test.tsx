import React from 'react';
import { render } from '@testing-library/react';
import IndexPage, { Head } from '../index';

describe('IndexPage', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<IndexPage />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('header meta data renders correctly', () => {
        const { asFragment } = render(<Head />);
        expect(asFragment()).toMatchSnapshot();
    });
});
