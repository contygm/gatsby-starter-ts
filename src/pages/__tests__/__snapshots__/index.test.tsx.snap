// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<DocumentFragment>
  <div
    id="app-container"
  >
    <nav
      aria-label="main navigation"
      class="navbar is-primary"
      role="navigation"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
          href="/"
        >
          <img
            alt="Logo"
            data-cy="nav-link-logo"
            layout="constrained"
            placeholder="blurred"
            src="../images/bulma-logo-white.png"
          />
        </a>
        <button
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger "
          data-cy="nav-hamburger"
          data-testid="siteNav"
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </button>
      </div>
      <div
        class="navbar-menu "
        id="siteNav"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            data-cy="nav-link-home"
            href="/"
          >
            Home
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-blog"
            href="/"
          >
            Blog
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-wiki"
            href="/"
          >
            Wiki
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-glossary"
            href="/404"
          >
            Glossary
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <div
            class="field nav-side-width"
          >
            <p
              class="control has-icons-left has-icons-right"
            >
              <input
                class="input is-rounded is-small"
                placeholder="Search..."
                type="text"
              />
              <span
                class="icon is-small is-left"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass "
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </p>
          </div>
        </div>
      </div>
    </nav>
    <main>
      <div
        class="content"
      >
        <section
          class="section pb-2"
        >
          <h1
            class="title is-1"
          >
            You're Home!
          </h1>
        </section>
        <section
          class="section pt-2"
        >
          <img
            alt="Logo"
            layout="constrained"
            placeholder="blurred"
            src="../images/xlarge.png"
          />
          <p>
            Ice cream danish cookie muffin soufflé lollipop. Bonbon sweet roll topping candy canes halvah pastry jelly-o. Candy canes candy chocolate bar shortbread gummi bears muffin cake bear claw icing.
          </p>
          <p>
            Candy canes chocolate cake caramels bonbon sesame snaps soufflé gummi bears powder chupa chups. Tiramisu oat cake shortbread tootsie roll candy gingerbread. Carrot cake halvah soufflé pudding macaroon marshmallow marzipan carrot cake bonbon.
          </p>
          <h2
            class="title is-2"
          >
            Second Heading
          </h2>
          <p>
            Chocolate cake cotton candy chupa chups marzipan topping. Caramels bonbon bonbon lollipop jujubes cupcake cookie. Cookie tiramisu cake pie pudding chocolate cake dessert carrot cake.
          </p>
          <p>
            Brownie sweet roll chupa chups cotton candy marshmallow pudding wafer. Donut macaroon danish topping pastry brownie carrot cake. Pudding chocolate tootsie roll bear claw toffee toffee bear claw. Biscuit tootsie roll sweet roll caramels gingerbread toffee.
          </p>
          <h3
            class="title is-3"
          >
            Third Heading
          </h3>
          <p>
            Chocolate cake cotton candy chupa chups marzipan topping. Caramels bonbon bonbon lollipop jujubes cupcake cookie. Cookie tiramisu cake pie pudding chocolate cake dessert carrot cake.
          </p>
          <p>
            Brownie sweet roll chupa chups cotton candy marshmallow pudding wafer. Donut macaroon danish topping pastry brownie carrot cake. Pudding chocolate tootsie roll bear claw toffee toffee bear claw. Biscuit tootsie roll sweet roll caramels gingerbread toffee.
          </p>
        </section>
      </div>
    </main>
    <footer
      class="footer"
    >
      <div
        class="columns mb-1 is-vcentered has-text-centered-mobile"
      >
        <div
          class="column"
        >
          <a
            class=""
            data-cy="footer-logo"
            href="/"
          >
            <figure
              class="image footer-logo"
            >
              <img
                alt="Logo"
                layout="constrained"
                placeholder="blurred"
                src="../images/bulma-logo-white.png"
                width="128"
              />
            </figure>
          </a>
        </div>
        <ul
          class="column has-text-centered"
          style="flex-basis: auto;"
        >
          <div>
            <li
              class="is-inline-desktop is-inline-tablet m-4 footer-link"
            >
              <a
                class="footer-link"
                data-cy="footer-link-Contact"
                href="/about#contact-us"
              >
                Contact
              </a>
            </li>
            <li
              class="is-inline-desktop is-inline-tablet m-4 footer-link"
            >
              <a
                class="footer-link"
                data-cy="footer-link-About"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="is-inline-desktop is-inline-tablet m-4 footer-link"
            >
              <a
                class="footer-link"
                data-cy="footer-link-Privacy"
                href="/legal/privacy-policy"
              >
                Privacy
              </a>
            </li>
            <li
              class="dropdown is-hoverable is-up"
            >
              <div
                class="dropdown-trigger"
              >
                <a
                  aria-controls="footer-dropdown-menu"
                  aria-haspopup="true"
                  class="m-4 footer-link"
                  data-cy="footer-link-legal"
                  href="/legal"
                >
                  Legal
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-up fa-sm px-2"
                    data-icon="angle-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="dropdown-menu"
                id="footer-dropdown-menu"
                role="menu"
              >
                <div
                  class="dropdown-content"
                >
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-copyright"
                      href="/legal/copyright"
                    >
                      Copyright
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-cookie"
                      href="/legal/cookie-policy"
                    >
                      Cookie Policy
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-terms"
                      href="/legal/terms-and-conditions"
                    >
                      Terms and Conditions
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-disclaimer"
                      href="/legal/disclaimer"
                    >
                      Disclaimer
                    </a>
                  </div>
                </div>
              </div>
            </li>
          </div>
        </ul>
        <div
          class="column"
        >
          <ul
            class="has-text-right has-text-centered-mobile footer-link"
          >
            <li
              class="footer-falink is-inline"
            >
              <a
                class="footer-link"
                data-cy="footer-falink"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star fa-xl "
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
            <li
              class="footer-falink is-inline"
            >
              <a
                class="footer-link"
                data-cy="footer-falink"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-mug-hot fa-xl "
                  data-icon="mug-hot"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
            <li
              class="footer-falink is-inline"
            >
              <a
                class="footer-link"
                data-cy="footer-falink"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-ghost fa-xl "
                  data-icon="ghost"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 384 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="footer-copyright has-text-centered has-text-grey-light is-size-7"
      >
        <p
          class="pt-3"
          data-testid="copyright"
        >
          © 
          <time>
            2022
          </time>
          , Built by Writer B. Author
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
